pkg/controller/utils/utils.go

```diff
@@ -439,6 +439,17 @@ func GetManagementClusterConnection(ctx context.Context, c client.Client) (*oper
 	return managementClusterConnection, nil
 }
 
+// GetNonClusterHost finds the NonClusterHost CR in your cluster.
+func GetNonClusterHost(ctx context.Context, cli client.Client) (*operatorv1.NonClusterHost, error) {
+	nonclusterhost := &operatorv1.NonClusterHost{}
+	err := cli.Get(ctx, DefaultTSEEInstanceKey, nonclusterhost)
```

Comment:
So, this function isn't following the pattern that the rest of the `GetX` functions follow (namely, they swallow IsNotFound errors).

Personally, I dislike how the other functions swallow the IsNotFound error and find them somewhat confusing. But we probably still want to be consistent with them to avoid further confusion.

Reply:
I checked that too, but our getX funcs are very inconsistent. Some swallow, others don't... :/

Reply:
:crying_cat_face:

Reply:
This function is updated to follow others. It will swallow `IsNotFound` errors.